src/lib/core/error_payload.h

```diff
@@ -0,0 +1,166 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include "trivia/util.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif /* defined(__cplusplus) */
+
+/** Single field of error payload. */
+struct error_field {
+	/** Data size. */
+	uint32_t size;
+	/** MessagePack field value. */
```

Comment:
It is mentioned in the `.c` file. But it does not matter much, because the 0 is not encoded into MessagePack data. External user of the field can just not care about it. As for hiding the `error_field` - I thought about it but decided to keep it here. Firstly it is used in `error.lua` anyway (although it is copy-pasted there). Secondly, I need it in a few places in the unit tests.

Reply:
OK, the problem will happen when external user tries to modify error_field, which must be prohibited.